plugins/inputs/modbus_gateway/convert.go

```diff
@@ -0,0 +1,117 @@
+/*
+ * These classes is a dynamic implementation of the go binary.ByteOrder interface that lets
+ * you specify a byte order consistent with how various modbus libraries express ordering:
+ *    [Modbus Poll](https://www.modbustools.com/poll_display_formats.html)
+ * Users of this class pass it to a bytes.Reader which should do the rest of the work, including
+ * filling in all the types that are not part of binary.ByteOrder
+ *
+ * Written by Christopher Piggott with the hope it will become standard in a replacement modbus
+ * library (some time in the future)
+ */
+
+package modbus_gateway
+
+import (
+	"encoding/binary"
+	"strings"
+)
+
+var byteOrderCache map[string]binary.ByteOrder
+
+func getOrCreateByteOrder(orderSpec string) binary.ByteOrder {
+	key := strings.ToUpper(orderSpec)
+	if byteOrderCache[key] != nil {
+		return byteOrderCache[key]
+	} else {
+		formatter, _ := CreateCustomByteOrder(key)
+		byteOrderCache[key] = formatter
+		return formatter
+	}
+}
+
+func CreateCustomByteOrder(orderSpec string) (CustomByteOrder, error) {
+	orderSpecUC := strings.ToUpper(orderSpec)
+	orderSpecLen := len(orderSpecUC)
+	orderSpecBytes := []byte(orderSpecUC)
+
+	converter := CustomByteOrder{
+		order: orderSpecUC,
+	}
+
+	i := 0
+	for ; i < orderSpecLen; i++ {
```

Comment:
I suggest  `for i := 0; i < orderSpecLen; i++ `

Reply:
No, i need i to pick up where it left off in the next for loop down

Reply:
I see, but at the final of the first loop always `i` be equal to `orderSpecLen`

you can use `orderSpecLen `in the next loop

`for ;  orderSpecLen < 8; i++`

Reply:
Right, thank you